### PR TITLE
Admin backend modules 

### PR DESCRIPTION
**Pull request by PrakharSMathur**
Issue no #3
Testing codes need to be checked and done.
Router files need authorisation update.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
```
Comment (the-lost-explorer) [inline index.js]: @shubham1172 I feel the file name should be changed.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
+var app = express();
+
+app.get('/', function(req, res){
+   res.send("Hello world!");
+});
+
+app.get('/hello', function(req, res){
```
Comment (the-lost-explorer) [inline index.js]: Unnecessary endpoint. Remove.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
+var app = express();
+
+app.get('/', function(req, res){
```
Comment (the-lost-explorer) [inline index.js]: Remove.

File: controllers/admin.controller.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    });
+}
+
+//Delete users
+exports.delete_users = (req,res) => {
+  User.findOneAndRemove({}, (err, deleted)=> {
+      if(err) {
+        res.status(200).send('error removing')
+      } else {
+       console.log(deleted);
```
Comment (the-lost-explorer) [inline controllers/admin.controller.js]: Remove log. Use logger.

File: controllers/admin.controller.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      username:req.body.username
+  },
+      { $set: { username: req.body.username ,
+          password: req.body.password,
+          profile_picture_path:req.body.profile_picture_path} },
+      {upsert: false}, (err, updated)=> {
+      if (err) {
+        res.status(200).send('error updating ')
+      } else {
+        console.log('updated with '+updated)
```
Comment (the-lost-explorer) [inline controllers/admin.controller.js]: Remove log. User logger.

File: controllers/admin.controller.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
+//Requirements
```
Comment (the-lost-explorer) [inline controllers/admin.controller.js]: Remove console.log() statements from the whole file.

File: index.js (hunk 3 above)
Comment (the-lost-explorer) [inline index.js]: Unnecessary endpoint.

File: index.js (hunk 2 above)
Comment (the-lost-explorer) [inline index.js]: Unnecessary endpoint.

File: package.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
   },
   "scripts": {
     "start": "node app.js",
-    "test": "./node_modules/.bin/mocha --recursive --exit"
+    "test": "mocha --recursive --exit"
```
Comment (the-lost-explorer) [inline package.json]:
```suggestion
    "test": "./node_modules/.bin/mocha --recursive --exit"
```

File: routes/admin.router.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+const express = require('express')
+const router = express.Router()
+const util = require('./util.route')
+const user_controller = require('../controllers/admin.controller')
+
+
+moduule.exports=router
```
Comment (the-lost-explorer) [inline routes/admin.router.js]:
```suggestion
module.exports=router 
```

File: test/appTest.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+
+const mongoose = require('mongoose');
+
+//tell mongoose to use es6 implementation of promises
+mongoose.Promise = global.Promise;
+
+mongoose.connect('mongodb://localhost/munverse');
+mongoose.connection
+    .once('open', () => console.log('Connected!'))
```
Comment (the-lost-explorer) [inline test/appTest.js]: Use logger.

File: test/appTest.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    .on('error', (error) => {
+        console.warn('Error : ',error);
```
Comment (the-lost-explorer) [inline test/appTest.js]: Use logger

File: test/read_test.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+//throws error finding a pre-inserted entry.
+
+const assert = require('assert');
+const Pokemon = require('../models/council.model');
```
Comment (the-lost-explorer) [inline test/read_test.js]: Pokemon? Please explain.

File: config.js [hunk 12]
```diff
@@ -1,6 +1,6 @@
 var config = {
-    port: process.env.PORT || 8080,
+    port: process.env.PORT || 8079,
```
Comment (shubham1172) [inline config.js]: Why did you change the port?
```suggestion
    port: process.env.PORT || 8080,
```

File: routes/admin.router.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+const express = require('express')
+const router = express.Router()
+const util = require('./util.route')
+const user_controller = require('../controllers/admin.controller')
+
```
Comment (shubham1172) [inline routes/admin.router.js]: Where are the routes?

File: test/appTest.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
+//MOCHA/CHAI tester for DB connection success
```
Comment (shubham1172) [inline test/appTest.js]: This file shouldn't exist.

**Comment by shubham1172** [comment]
Closing due to inactivity.
